### PR TITLE
MMDP-458 Implement source of funding dropdown

### PR DESCRIPTION
**Pull request by Ruiru11**
**What does this PR do?**

- Allow an admin to manage the source of funding dropdown

**description of Task to be completed?**

- Admin should create a source of funding
- Admin should be able to delete a source of funding
- Admin should be able to edit a source of funding
- Admin should not be able to delete a source of funding if it is assigned to a returnee service

**How should this be manually tested?**

- run `git fetch` and then run `git checkout ft-funding-dropdown-MMDP-458`
- run `yarn run dev-server`
- on your browser navigate to `http://localhost:1234/dropdowns/funding-source`

**Any background context you want to provide?**

#### NONE

**What are the relevant Jira issues?**

[MMDP-458](https://viisaus.atlassian.net/browse/MMDP-458)

**Screenshots (if appropriate)**

image

**Questions:**

#### NONE

### CODE REVIEW

File: client/containers/DropDowns/SourceOfFunding/index.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    // validate stateName duplicates
+    const editData = filterDropdowns(fundings, hasId);
+    const addData = filterDropdowns(fundings, hasNoId);
+
+    editData.map((item) => {
+      function condition(fundingData) {
+        return fundingData.sourceOfFundingName === item.sourceOfFundingName;
+      }
+      const duplicates = addData.filter(condition);
+      // console.warn(duplicates)
```
Comment (p8ul) [inline client/containers/DropDowns/SourceOfFunding/index.js]: Please remove commented code.

File: client/store/sagas/dropdowns/funding.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    const message = response
+      ? response.data.message
+      : 'Source of funding added successfully';
+    toastr.success(message);
+  } catch (error) {
+    yield put(actions.addFundingFailure({}));
+    const message = error.response
+      ? error.response.data.error.data
+      : 'Error adding source of funding';
+    toastr.warning(message);
```
Comment (eduhmik) [inline client/store/sagas/dropdowns/funding.js]: You can consider using yield call (fn, args) as @p8ul proposed in the slack channel today here to make it easier for you to test. Thanks and keep up the good job.

File: client/containers/DropDowns/SourceOfFunding/index.js (hunk 1 above)
Comment (Ruiru11) [reply client/containers/DropDowns/SourceOfFunding/index.js]: thanks i have worked on it

File: client/store/sagas/dropdowns/funding.js (hunk 2 above)
Comment (Ruiru11) [reply client/store/sagas/dropdowns/funding.js]: thanks @eduhmik i have worked on that

File: client/components/DropDowns/SourceOfFunding/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            <DropdownForm
+              // eslint-disable-next-line
+              key={index}
+              item={item}
+              inputs={inputs || []}
+              header={header}
+              label={content}
+              editAState={editFunding}
+              deleteAState={deleteAFunding}
+              id="fs"
```
Comment (p8ul) [inline client/components/DropDowns/SourceOfFunding/index.js]: Please override the common component with a `className` The id attribute specifies a unique id for an HTML element (the value must be unique within the HTML document).

also `fs` is not descriptive.
Screenshot 2019-04-02 at 12 24 32

File: client/components/common/Form/DropdownForm/index.js [hunk 4]
```diff
@@ -49,7 +50,7 @@ const DropdownForm = ({
         );
       })}
 
-      <div className="dropdowns__actions">
+      <div className="dropdowns__actions" id={id || ''}>
```
Comment (p8ul) [inline client/components/common/Form/DropdownForm/index.js]: From the over feedback do something like `<div className=`dropdowns_actions ${className}`>` and pass the className as a prop to the common component. Id should be unique in HTML document

**Pull request merged**